src/CFamily.UnitTests/SystemAbstractions/ProcessWrapperTests.cs

```diff
@@ -0,0 +1,212 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarLint.VisualStudio.CFamily.SystemAbstractions;
+using SonarLint.VisualStudio.Integration.UnitTests.Helpers;
+
+namespace SonarLint.VisualStudio.CFamily.UnitTests.SystemAbstractions
+{
+    [TestClass]
+    public class ProcessWrapperTests
+    {
+        public TestContext TestContext { get; set; }
+
+        [TestMethod]
+        public void ProcessFactory_Start_ReturnsWrappedProcess()
+        {
+            var exeName = WriteBatchFileForTest(TestContext, "@echo hello world");
+
+            IProcess wrappedProcess = null;
+            var processStartInfo = CreateProcessStartInfo(exeName);
+            var testSubject = new ProcessFactory();
+
+            try
+            {
+                wrappedProcess = testSubject.Start(processStartInfo);
+
+                wrappedProcess.Should().NotBeNull();
+                wrappedProcess.StartInfo.Should().BeSameAs(processStartInfo);
+            }
+            finally
+            {
+                if (!wrappedProcess.HasExited)
+                {
+                    wrappedProcess?.Kill();
```

Comment:
This can throw. We do a try-catch in [similar tests](https://github.com/SonarSource/sonarlint-visualstudio/blob/master/src/TypeScript.UnitTests/EslintBridgeClient/EslintBridgeProcessTests.cs#L69) but maybe the wrapper can do that instead.

Reply:
I'll add a try-catch here and keep the wrapper as a simple passthrough for now.